api/app/serializers/v1/concerns/project_serializer.rb

```diff
@@ -24,6 +24,9 @@ module ProjectSerializer
         typed_attribute :recently_updated, Types::Bool.meta(read_only: true), &:recently_updated?
         typed_attribute :updated, Types::Bool.meta(read_only: true), &:updated?
         typed_attribute :avatar_styles, Types::Serializer::Attachment.meta(read_only: true)
+        typed_attribute :avatar_alt_text, Types::String.optional do |object, _params|
+          object.avatar.metadata["alt_text"] unless object.avatar.nil?
```

Comment:
When you find yourself writing method calls like this that involve a lot of deep navigation, that's usually a sign you want to use `delegate`.

Shrine also helps us here, because using `add_metadata` will automatically define a reader on the `UploadedFile` instance.

in `app/models/concerns/attachments.rb`, in the `manifold_has_attached_file` method, after `include AttachmentUploader::Attachment.new(field)`, add the following line:

```ruby
delegate :alt_text, to: field, prefix: true, allow_nil: true
```

That will create methods like `hero_alt_text`, `avatar_alt_text` etc that already safely navigate null attachments because of the `allow_nil: true` option. You can then remove all these blocks from this and all the similar ones. It will just read the method directly on the model since it's the same name.

Reply:
Nifty!